packages/itwinui-react/tsconfig.test.json

```diff
@@ -0,0 +1,5 @@
+{
+  "extends": "./tsconfig.json",
+  // TODO: Test all the files, and not just the Table files or just the build files.
+  "include": ["src/core/Table/**/*"]
```

Comment:
is this path correct? the test was moved out of `core`

Reply:
i also the think the comment above can be removed. we will likely never extensive typescript tests for all components, because we shouldn't need to.

even this `react-table.ts` will be removed after we have tanstack-table working

Reply:
> is this path correct? the test was moved out of `core`

https://github.com/iTwin/iTwinUI/pull/1514/commits/7c1ca720e1d30f863057c2f15f333b637e42784b: I added `src/react-table/*` to the `include` so that apart from all Table files, `react-table.types-test.tsx` will also be tested.

> i also the think the comment above can be removed. we will likely never add extensive typescript tests for all components, because we shouldn't need to.
> 
> even this `react-table.ts` will be removed after we have tanstack-table working

I feel at least for Table, it would be good to check if there are any type errors in all files, and not just the build files. This is because the Table unit test file, for example, might pass but if there is a type error there, we will not know to fix the Table types.

I feel the same applies to other components too. Is there a downside to testing the types for Table and other components?

Reply:
> I feel at least for Table, it would be good to check if there are any type errors in all files, and not just the build files. This is because the Table unit test file, for example, might pass but if there is a type error there, we will not know to fix the Table types.

jest will already fail if there is a type error in the unit test, so it doesn't need to be manually checked again

> Is there a downside to testing the types for Table and other components?

yes, it's more code maintain with no benefit. we shouldn't ever have to write such complex types that they require their own tests

Reply:
> jest will already fail if there is a type error in the unit test, so it doesn't need to be manually checked again

I had intentionally placed a type error in the Table unit test file. When running `yarn test:unit Table.test.tsx`, all tests had passed and no errors were. I reconfirmed this now too, the type error was not reported by jest

> > Is there a downside to testing the types for Table and other components?
> 
> yes, it's more code maintain with no benefit. we shouldn't ever have to write such complex types that they require their own tests

To clarify, I am not suggesting to write `react-table.types-test.tsx` for all components. What I am suggesting is to type check all existing files and not just the files in the build pipeline. That way, we were ever do some type error in the unit test or if we ever do a PR in a component that messes up the types for only the way the component is being used in the unit test, we will have a way of catching it.

Reply:
we can look into type checking unit tests, sure.

lets continue this discussion some other time, as it's going outside the scope of this PR.